docs/configure/sidebar-and-urls.md

```diff
@@ -26,76 +26,100 @@ If you’d prefer to show top-level nodes as folders rather than roots, you can
 
 <!-- prettier-ignore-end -->
 
-## Automatically generate titles
+## Permalink to stories
 
-With CSF 3.0 files, you can choose to leave out a title and let it be inferred from the story's path on disk instead:
+By default, Storybook generates an `id` for each story based on the component title and the story name. This `id` in particular is used in the URL for each story, and that URL can serve as a permalink (primarily when you [publish](../sharing/publish-storybook.md) your Storybook).
+
+Consider the following story:
 
 <!-- prettier-ignore-start -->
 
 <CodeSnippets
   paths={[
-    'common/component-story-auto-title.js.mdx',
+    'common/foo-bar-baz-story.js.mdx',
   ]}
 />
 
 <!-- prettier-ignore-end -->
 
-This is controlled by the way your stories are configured:
+Storybook's ID-generation logic will give this the `id` `foo-bar--baz`, so the link would be `?path=/story/foo-bar--baz`.
+
+It is possible to manually set the story's id, which is helpful if you want to rename stories without breaking permalinks. Suppose you want to change the position in the hierarchy to `OtherFoo/Bar` and the story name to `Moo`. Here's how to do that:
 
 <!-- prettier-ignore-start -->
 
 <CodeSnippets
   paths={[
-    'common/component-story-configuration.js.mdx',
+    'common/other-foo-bar-story.js.mdx',
   ]}
 />
 
 <!-- prettier-ignore-end -->
 
-Given this configuration, the stories file `../src/components/MyComponent.stories.tsx` will get the title `components/MyComponent`.
+Storybook will prioritize the `id` over the title for ID generation if provided and prioritize the `story.name` over the export key for display.
 
-You can further customize the generated title by modifying the stories configuration.
+## CSF 3.0 auto-titles
+
+Storybook 6.4 introduced CSF 3.0 as an experimental feature, allowing you to write stories more compactly. Suppose you're already using this format to write your stories. In that case, you can omit the `title` element from the default export and allow Storybook automatically infer it based on the file's physical location. For example, given the following configuration and story:
```

Comment:
When you say `CSF 3.0` and refers with `using this format` I really feel like there should be some link or reference to that. Also, auto title is not necessarily tied to CSF 3.0! It's a logic in the parsing of CSF files that should work with CSF2 (or CSF1) stories too